### PR TITLE
feat(zupass-gatekeeper): added zupass gatekeeper support

### PR DESCRIPTION
**Pull request by crisgarner**
Added support to Zupass gatekeeper

### CODE REVIEW

**Review by kittybest — COMMENTED** [review]
can run `pnpm lint:ts` first to avoid the github action error

**Comment by crisgarner** [comment]
> can run `pnpm lint:ts` first to avoid the github action error

Thanks! fixed the issues

File: packages/interface/src/config.ts [hunk 1]
```diff
@@ -115,6 +115,15 @@ export const eas = {
   },
 };
 
+export const zupass = {
+  pcdType: "eddsa-ticket-pcd",
+  publicKey: [
+    "1ebfb986fbac5113f8e2c72286fe9362f8e7d211dbc68227a468d7b919e75003",
```
Comment (ctrlc03) [inline packages/interface/src/config.ts:121]: are these constants for all eddsa-ticket-pcd?

File: packages/interface/src/utils/pcd.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/* eslint-disable */
```
Comment (ctrlc03) [inline packages/interface/src/utils/pcd.ts:1]: are none of these exported by any of zupass libraries?

File: packages/interface/src/contexts/Maci.tsx [hunk 3]
```diff
@@ -209,6 +229,13 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
     setSemaphoreIdentity(newSemaphoreIdentity);
   }, [address, signatureMessage, signMessageAsync, setMaciPrivKey, setMaciPubKey, setSemaphoreIdentity]);
 
+  const generateZupassProof = useCallback(
```
Comment (ctrlc03) [inline packages/interface/src/contexts/Maci.tsx:232]: maybe a better name is `storeZupassProof`?

**Review by ctrlc03 — COMMENTED** [review]
Thanks left some comments

File: packages/interface/src/utils/pcd.ts (hunk 2 above)
Comment (crisgarner) [reply packages/interface/src/utils/pcd.ts:1]: Just checked their codebase and seems like they use some of this code internally, not sure why they don't export it,  will ask around why, if not will submit a PR to them

File: packages/interface/src/config.ts (hunk 1 above)
Comment (crisgarner) [reply packages/interface/src/config.ts:121]: Currently, they are, in theory at some point they would change who could sign pcds.

Comment (ctrlc03) [reply packages/interface/src/config.ts:121]: oh ok

**Review by ctrlc03 — APPROVED** [review]
Thanks

**Pull request merged**